src/content/en/updates/2018/07/page-lifecycle-api.md

```diff
@@ -0,0 +1,962 @@
+    logStateChange('frozen');
+  } else {
+    // If the event's persisted property is not `true` the page is
+    // about to be unloaded.
+    logStateChange('terminated');
+  }
+}, {capture: true});
+```
+
+<aside class="warning">
+  <strong>Warning!</strong>
+  This code yields different results in different browsers, as the order
+  (and reliability) of events has not been consistently implemented. To learn
+  how best to handle these inconsistencies see managing cross-browsers
+  differences.
+</aside>
+
+The above code does three primary things:
+
+1. It sets the initial state using the `getState()` function.
+2. It defines a function that accepts a next state and, if there's a change,
+   logs the state changes to the console.
+3. It adds capturing events listeners for all necessary lifecycle events, which
+   in turn call `logStateChange()`, passing in the next state.
+
+One thing to note about the above code is that all the event listeners are added
+to `window` and they all pass
+[`{capture: true}`](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#Syntax).
+There are three reasons for this:
+
+1. Not all Page Lifecycle events have the same target. `pagehide`, and
+   `pageshow` are fired on `window`; `visibilitychange`, 'freeze', and
+   'resume' are fired on `document`, and 'focus' and 'blur' are fired on their
+   respective DOM elements.
+2. Most of these events do not bubble, which means it's impossible to add
+   non-capturing event listeners to a common ancestor element and observe all
+   of them.
+3. The capture phase executes before the target or bubble phases, so adding
+   listeners there helps ensures they run before other code can cancel them.
+
+### Managing cross-browsers differences {: #managing-cross-browsers-differences}
+
+The chart in the beginning of this article outlines the state and event flow
+according to the Page Lifecycle API. But since this API has just been
+introduced, the new events and DOM APIs have not been implemented in all
+browsers.
+
+Furthermore, the events that are implemented in all browsers today are not
+implemented consistently. For example:
+
+1. Some browsers do not fire a `blur` event when switching tabs. This means
+   (contrary to the diagram and tables above) a page could go from the active
+   state to the hidden state without going through passive first.
+2. Several browsers implement a page navigation cache (also known as a
+   back-forward cache), and the Page Lifecycle API classifies cached pages as
+   being in the frozen state. Since this API is brand new, these browsers do not
+   yet implement the `freeze` and `resume` events, though this state can still
+   be observed via the `pagehide` and `pageshow` events.
+3. Older versions of Internet Explorer (10 and below) do not implement the
+   `visibilitychange` event.
+4. The ordering of when `pagehide` and `visibilitychange` events are dispatched
+   for unloading pages has
```

Comment:
Change to 'The  dispatch order of the `pagehide` and `visibilitychange` events has changed.

Reply:
Done.